### ISSUE TITLE
File output plugin should support File Rollover

### ISSUE THREAD

**Issue by kprasad99** [fluent/fluent-bit#1687]
**Is your feature request related to a problem? Please describe.**
Current `File` output plugin doesn't have mechanism for File Rollover,  
**Describe the solution you'd like**
* Would like to have Rollover mechanism based on,

1.  Time.
2. File size.
* The naming convention should contain time stamp file was created.
* Though good to have, there should be housing keeping activity, to clear archived logs, May be we can ignore this. this can be managed by separate application/docker which could be better handled by K8s cron jobs,
* Log message should be complete, shouldn't be partial in case of file size based roll over.
 
**Additional context**

[Stackoverflow question](https://stackoverflow.com/questions/58498040/configure-fluent-bit-file-output-plugin-set-file-rollover/58577944#58577944)

**Comment by nigels-com** [comment]
The general idea is a circular buffer that limits logging to a certain total size or maximum age?

If the service is stopped and started, do we expect those limits to be respected?

**Comment by kprasad99** [comment]
@nigels-com Probably shouldn't, the out plugin should be as light as possible.

**Comment by nigels-com** [comment]
So when the output file reaches a certain size or age the plugin should roll over to a fresh output file?  Leave the responsibility elsewhere for cleaning up?

**Comment by kprasad99** [comment]
@nigels-com Yes, we can easily schedule a job for cleaning, anyways platforms such as kubernetes provides one such.

**Comment by kprasad99** [comment]
@nigels-com  any progress?

**Comment by nigels-com** [comment]
@kprasad99 I'm tempted to take a crack at it (have a look) but I already have a few things in progress.
From the discussion the scope does seem modest and doable. (Famous last words!)

**Comment by epsteina16** [comment]
Reviving this issue. It would be great to have this as a feature for the file output plugin.

**Comment by ebberholmes** [comment]
I am also interested in this feature of file output-plugin.

**Comment by MattJud** [comment]
I also need this feature, especially for OpenSearch implementation. I don't need the fluenbit to handle the rollover, but I just need that when fluentbit created a new index in OpenSearch, it attach the required rollover property to the index
```
PUT log-000001
{
  "aliases": {
    "log": {
      "is_write_index": true
    }
  }
}
```
As seen in the example above, I require that when fluentbit create the index in OpenSearch, I need that aliases content to be attach to the payload
